models/user_test.go

```diff
@@ -28,8 +27,23 @@ func TestHashPassword(t *testing.T) {
 	assert.Equal(t, nil, result)
 }
 
+func TestCheckPasswordWithValidPassword(t *testing.T) {
+	hashedPassword, _ := hashPassword("password")
+	result := CheckPassword(string(hashedPassword), "password")
+
+	assert.Equal(t, nil, result)
+}
+
+func TestCheckPasswordWithInvalidPassword(t *testing.T) {
+	hashedPassword, _ := hashPassword("password")
+	result := CheckPassword(string(hashedPassword), "drowssap")
+
+	assert.NotEqual(t, nil, result)
+}
```

Comment:
Why are these tests defined before `DBTestSuite` and `SetupTest`? I would be more consistent to keep all tests below these basic blocks/ Now it fees a bit messy.

Reply:
Because all the defined before `DBTestSuite` test cases is not involve to database. I think it is better to keep the test cases that only involve to database define in `DBTestSuite` so `SetupTest` won't setup unnecessary stuffs to test cases that don't use them.

What do you think?

Reply:
> Because all the defined before DBTestSuite test cases are not involve to database. 

That kind of makes sense but I am not sure in what order Gin execute the tests 🤔 So does placing them before the DBTestSuite really affect the way tests are run or not?

Reply:
I cannot find any resource for this either.
I just group them to make it more easier to read.

Reply:
Not sure about making this as a convention though. I would prefer to keep the test setup at the top and then have all tests below it. 

```
# Now

setup
some tests
DB setup
more tests

# More standard

setup
DB setup
all tests

```

The tests which don't use the DB would not take the DB param which I think is readable enough IMO. It's a matter to all agree on obviously.